pybuildkite/client.py

```diff
@@ -52,8 +53,15 @@ def request(
 
         query_params = self._clean_query_params(query_params or {})
 
+        headers_with_auth = headers
         if self.access_token:
-            query_params["access_token"] = self.access_token
+            if headers_with_auth is None:
+                headers_with_auth = {}
+            else:
+                # copy the dict so that the below logic for determining if json should
+                # be returned is correct
+                headers_with_auth = dict(headers)
```

Comment:
Could we simplify this logic by having `headers` defaulted to a dict `{}` in all of our REST action methods and this request method? Then we can just check `if self.access_token and not headers is None:` and add https://github.com/pyasi/pybuildkite/pull/42/files#diff-96a8e9488db458e22181d770181dff46R64 directly to `headers`

Reply:
Good idea.

I think my latest commit addresses your ask, but I'm not sure if there is a cleaner way to keep backwards compatibility with the logic of "if any headers are passed in, return `response.json()`"(https://github.com/pyasi/pybuildkite/pull/42/files#diff-96a8e9488db458e22181d770181dff46R87)